content/en/blog/_posts/2024-12-11-volume-group-snapshot-beta.md

```diff
@@ -0,0 +1,373 @@
+
+```yaml
+NAME                                                                        READYTOUSE   SOURCEPVC   RESTORESIZE   SNAPSHOTCONTENT                                                                AGE
+snapshot-0962a745b2bf930bb385b7b50c9b08af471f1a16780726de19429dd9c94eaca0   true         pvc-0       100Mi         snapcontent-0962a745b2bf930bb385b7b50c9b08af471f1a16780726de19429dd9c94eaca0   16m
+snapshot-da577d76bd2106c410616b346b2e72440f6ec7b12a75156263b989192b78caff   true         pvc-1       100Mi         snapcontent-da577d76bd2106c410616b346b2e72440f6ec7b12a75156263b989192b78caff   16m
+```
+
+#### Importing an existing group snapshot with Kubernetes
+
+To import a pre-existing volume group snapshot into Kubernetes, you must also import
+the corresponding individual volume snapshots.
+
+Identify the individual volume snapshot handles, manually construct a
+VolumeSnapshotContent object first, then create a VolumeSnapshot object pointing to
+the VolumeSnapshotContent object. Repeat this for every individual volume snapshot.
+
+Then manually create a VolumeGroupSnapshotContent object, specifying the
+volumeGroupSnapshotHandle and individual volumeSnapshotHandles already existing
+on the storage system.
+
+```yaml
+apiVersion: groupsnapshot.storage.k8s.io/v1beta1
+kind: VolumeGroupSnapshotContent
+metadata:
+  name: static-group-content
+spec:
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
+  source:
+    groupSnapshotHandles:
+      volumeGroupSnapshotHandle: e8779136-a93e-11ef-9549-66940726f2fd
+      volumeSnapshotHandles:
+      - e8779147-a93e-11ef-9549-66940726f2fd
+      - e8783cd0-a93e-11ef-9549-66940726f2fd
+  volumeGroupSnapshotRef:
+    name: static-group-snapshot
+    namespace: default
+```
+
+After that create a VolumeGroupSnapshot object pointing to the VolumeGroupSnapshotContent
+object.
+
+```yaml
+apiVersion: groupsnapshot.storage.k8s.io/v1beta1
+kind: VolumeGroupSnapshot
+metadata:
+  name: static-group-snapshot
+  namespace: default
+spec:
+  source:
+    volumeGroupSnapshotContentName: static-group-content
+```
+
+### How to use group snapshot for restore in Kubernetes
+
+At restore time, the user can request a new PersistentVolumeClaim to be created from
+a VolumeSnapshot object that is part of a VolumeGroupSnapshot. This will trigger
+provisioning of a new volume that is pre-populated with data from the specified
+snapshot. The user should repeat this until all volumes are created from all the
+snapshots that are part of a group snapshot.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc0-restore
+  namespace: demo-namespace
+spec:
+  storageClassName: csi-sc
+  dataSource:
+    name: snapshot-0962a745b2bf930bb385b7b50c9b08af471f1a16780726de19429dd9c94eaca0
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 100Mi
```

Comment:
```suggestion
      storage: 100Mi # must be enough storage to fit the existing snapshot
```
?

I presume that someone could make an admission controller that fills in the storage request for you based on the snapshot, but I also presume this is out of scope for the announcement.

Reply:
Yes